jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/DataprocInfo.java

```diff
@@ -0,0 +1,188 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.dataproc.jdbc;
+
+import static com.google.cloud.dataproc.jdbc.HiveUrlUtils.checkUrl;
+
+import com.google.api.gax.rpc.ApiException;
+import com.google.api.gax.rpc.StatusCode;
+import com.google.cloud.dataproc.v1beta2.Cluster;
+import com.google.cloud.dataproc.v1beta2.ClusterControllerClient;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/** Helper class to get cluster info through Dataproc API. */
+public class DataprocInfo {
+    private static final String HIVE_PROTOCOL = "jdbc:hive2://";
+    private final ClusterControllerClient clusterControllerClient;
+    private final HiveJdbcConnectionOptions params;
+
+    // Constructor dependency injection
+    public DataprocInfo(HiveJdbcConnectionOptions params, ClusterControllerClient controller) {
+        this.params = params;
+        this.clusterControllerClient = controller;
+    }
+
+    /**
+     * Format client passed in url that is accepted by Dataproc to be Hive acceptable format.
+     * jdbc:hive2://<host>:<port>/<dbName>;transportMode=http;httpPath=<http_endpoint>;<otherSessionConfs>?<hiveConfs>#<hiveVars>
+     *
+     * @return
+     * @throws InvalidURLException
+     */
+    public String toHiveJdbcUrl() throws SQLException {
+        String hiveJdbcURL =
+                String.format(
+                        "%s%s:%d/%s;transportMode=%s;httpPath=%s",
+                        HIVE_PROTOCOL,
+                        getHost(),
+                        params.port(),
+                        params.dbName(),
+                        params.transportMode(),
+                        params.httpPath());
+        if (params.otherSessionConfs() != null) {
+            hiveJdbcURL = String.format("%s;%s", hiveJdbcURL, params.otherSessionConfs());
+        }
+        if (params.hiveConfs() != null) {
+            hiveJdbcURL = String.format("%s?%s", hiveJdbcURL, params.hiveConfs());
+        }
+        if (params.hiveVars() != null) {
+            hiveJdbcURL = String.format("%s#%s", hiveJdbcURL, params.hiveVars());
+        }
+        return hiveJdbcURL;
+    }
+
+    /**
+     * Use Dataproc client libraries to get cluster.
+     *
+     * @return name of the cluster's master node
+     * @throws InvalidURLException
+     */
+    public String getHost() throws SQLException {
+        String host =
+                params.clusterName() == null
```

Comment:
Seems like indentation is wrong here. May you reformat with Google Format?

Reply:
This is how `java -jar google-java-format-1.8-all-deps.jar` formats it. 😂

Reply:
Interesting, I thought that it will indent 4 spaces here, but if it wants 8, let's leave it as is :)